ResearchKit/Common/ORKImageCaptureView.m

```diff
@@ -140,10 +157,18 @@ - (UIImage *)capturedImage {
 - (void)setError:(NSError * __nullable)error {
     _error = error;
     _headerView.alpha = error==nil ? 0 : 1;
-    _headerView.instructionLabel.text = [error.userInfo valueForKey:NSLocalizedDescriptionKey];
-    if (error && self.continueTitle) {
-        _continueSkipContainer.continueButtonItem.title = self.continueTitle;
-        _continueSkipContainer.skipButtonItem = nil;
+    _headerView.instructionLabel.text = error==nil ? nil : [error.userInfo valueForKey:NSLocalizedDescriptionKey];
+    
+    if (error) {
+        // Show skip, if available, and hide the template and continue/capture button
+        _previewView.templateImageHidden = YES;
+        _continueSkipContainer.continueButtonItem = nil;
+        _continueSkipContainer.skipButtonItem = _skipButtonItem;
+    } else {
+        // Reset to running state
+        _previewView.templateImageHidden = NO;
```

Comment:
Can it ever be true that we set the error to nil, and that there is a captured image present?

I think this code and the code in setCapturedImage: should be factored out so there's just one set of conditionals applied, i.e., so the logic depends only on state and not on transient factors.